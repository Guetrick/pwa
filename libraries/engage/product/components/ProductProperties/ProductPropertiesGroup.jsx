import React from 'react';
import PropTypes from 'prop-types';
import { subgroup } from './ProductProperties.style';

/**
 * @param {Object} props The component props.
 * @return {JSX}
 */
const ProductPropertiesGroup = ({ group }) => (
  <tr>
    <td colSpan="2" className={subgroup}>{group}</td>
  </tr>
);

ProductPropertiesGroup.propTypes = {
  group: PropTypes.string.isRequired,
};

export default React.memo(ProductPropertiesGroup);
