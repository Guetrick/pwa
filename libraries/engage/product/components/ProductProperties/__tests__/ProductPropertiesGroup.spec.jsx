import React from 'react';
import { shallow } from 'enzyme';
import ProductPropertiesGroup from '../ProductPropertiesGroup';

describe('<ProductPropertiesGroup />', () => {
  it('should render a group title row', () => {
    const wrapper = shallow(<ProductPropertiesGroup group="Test" />);
    expect(wrapper.find('tr').length).toEqual(1);
    expect(wrapper.find('td').length).toEqual(1);
    expect(wrapper.find('td').text()).toEqual('Test');
    expect(wrapper).toMatchSnapshot();
  });
});
