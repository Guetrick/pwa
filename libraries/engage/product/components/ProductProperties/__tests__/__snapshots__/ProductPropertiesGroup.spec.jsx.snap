// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesGroup /> should render a group title row 1`] = `
<tr>
  <td
    className={
      Object {
        "data-css-1o9kcbq": "",
      }
    }
    colSpan="2"
  >
    Test
  </td>
</tr>
`;
